code/game/objects/structures/window_frames.dm

```diff
@@ -0,0 +1,485 @@
+	user.do_attack_animation(src)
+	user.changeNext_move(CLICK_CD_MELEE)
+	user.visible_message(span_warning("[user] mangles [src]."), null, null, COMBAT_MESSAGE_RANGE)
+	if(!try_shock(user, 70))
+		take_damage(20, BRUTE, MELEE, 1)
+
+/obj/structure/window_frame/wirecutter_act(mob/living/user, obj/item/tool)
+	add_fingerprint(user)
+	if(try_shock(user, 100))
+		return
+	if(!has_grille)
+		return
+	if(!tool.use_tool(src, user, 0, volume = 50))
+		return
+	tool.play_tool_sound(src, 100)
+	to_chat(user, "<span class='notice'>You cut the grille on [src].</span>")
+	has_grille = FALSE
+	update_appearance()
+	return ITEM_INTERACT_SUCCESS
+
+
+/obj/structure/window_frame/welder_act(mob/living/user, obj/item/tool)
+	. = ..()
+	add_fingerprint(user)
+	if(atom_integrity >= max_integrity)
+		to_chat(user, span_warning("[src] is already in good condition!"))
+		return
+	if(!tool.tool_start_check(user, amount = 0))
+		return
+
+	to_chat(user, span_notice("You begin repairing [src]..."))
+	if(!tool.use_tool(src, user, 40, volume = 50))
+		return
+
+	atom_integrity = max_integrity
+	to_chat(user, span_notice("You repair [src]."))
+	update_appearance()
+	return ITEM_INTERACT_SUCCESS
+
+///creates a window from the typepath given from window_type, which is either a glass sheet typepath or a /obj/structure/window subtype
+/obj/structure/window_frame/proc/create_structure_window(window_material_type, start_anchored = TRUE)
+	var/obj/structure/window/our_window
+
+	if(ispath(window_material_type, /obj/structure/window))
+		our_window = new window_material_type(loc)
+		if(!our_window.fulltile)
+			stack_trace("Window frames can't use non fulltile windows!")
+
+	//window_material_type isnt a window typepath, so check if its a material typepath
+	if(ispath(window_material_type, /obj/item/stack/sheet/glass))
+		our_window = new /obj/structure/window/fulltile(loc)
+
+	if(ispath(window_material_type, /obj/item/stack/sheet/rglass))
+		our_window = new /obj/structure/window/reinforced/fulltile(loc)
+
+	if(ispath(window_material_type, /obj/item/stack/sheet/plasmaglass))
+		our_window = new /obj/structure/window/plasma/fulltile(loc)
+
+	if(ispath(window_material_type, /obj/item/stack/sheet/plasmarglass))
+		our_window = new /obj/structure/window/reinforced/plasma/fulltile(loc)
+
+	if(ispath(window_material_type, /obj/item/stack/sheet/titaniumglass))
+		our_window = new /obj/structure/window/reinforced/shuttle(loc)
+
+	if(ispath(window_material_type, /obj/item/stack/sheet/plastitaniumglass))
+		our_window = new /obj/structure/window/reinforced/plasma/plastitanium(loc)
+
+	if(ispath(window_material_type, /obj/item/stack/sheet/paperframes))
+		our_window = new /obj/structure/window/paperframe(loc)
+
+	if(!start_anchored)
+		our_window.set_anchored(FALSE)
+		our_window.state = WINDOW_OUT_OF_FRAME
+
+	our_window.update_appearance()
+
+/obj/structure/window_frame/attackby(obj/item/attacking_item, mob/living/user, params)
```

Comment:
It could use item_interaction instead of attackby.

Reply:
uhhhh how

Reply:
Eh, it isn't really necessary anyway, it's more or less a refactored part of the attack proc-chain Melbert did to deal with a couple attackby overrides and bs. I'll just resolve this.